charts/simple-app/templates/deployment.yaml

```diff
@@ -140,7 +140,18 @@ spec:
         {{- with $.Values.volumes }}{{ toYaml . | nindent 8 }}{{ end }}
         {{- with $.Values.volumesString }}{{ tpl . $ | nindent 8 }}{{ end }}
       {{- end }}
+      initContainers:
+        {{- with $.Values.initContainers}}
```

Comment:
Move the `{{- with ...` to line 142, so that if no `initContainers` are provided, then we don't even create the `initContainer` key. Otherwise you end up with templating like this:

```yaml
...
initContainers:
containers:
  - ...
```

That kind of templating causes tools like ArgoCD to throw a fit in certain situations.